server_state.go

```diff
@@ -191,4 +205,13 @@ type OwnerVoucherPersistentState interface {
 
 	// Voucher retrieves a voucher by GUID.
 	Voucher(context.Context, protocol.GUID) (*Voucher, error)
+
+	// VoucherStatus returns the status of a voucher by GUID.
+	VoucherStatus(context.Context, protocol.GUID) (VoucherStatus, error)
+}
+
+// OwnerDevicePersistentState manages device vouchers that have been onboarded.
+type OwnerDevicePersistentState interface {
+	// Device retrieves a device voucher by GUID.
+	Device(context.Context, protocol.GUID) (*Voucher, error)
```

Comment:
This is logically part of `ManufacturerVoucherPersistentState` and the name `Device()` doesn't make sense. it returns a voucher, just like `Voucher()`. I guess it would be `UnextendedVoucher()`?

Reply:
One thing that's a little weird: this function is NOT REQUIRED for the actual implementation, though it IS REQUIRED for fdotest. So I guess we should make a note in the godoc that it may be left unimplemented, but will cause fdotest to crash.

Reply:
The core question is this: in the FDO protocol, a device is in the same state whether it has just completed DI (auto extension is not part of the protocol) or just completed TO2 (without the credential reuse protocol). However, should the library treat them the same? A device that has been onboarded is very different to an Owner Service than a device that has been initialized at a manufacturer and then never extended for onboarding.